src/DataFrame/IO/Parquet/Thrift.hs

```diff
@@ -287,7 +287,6 @@ readFileMetaData metadata metaDataBuf bufferPos lastFieldId fieldStack = do
                         then readVarIntFromBuffer @Int metaDataBuf bufferPos
                         else return $ fromIntegral ((sizeAndType `shiftR` 4) .&. 0x0f)
 
-                let elemType = toTType sizeAndType
```

Comment:
This is actually supposed to be used to check that the elem type is what you expect. So if the metadata says this is a list of string but you find yourself in a list of int you should panic. I didn't put much effort into it since this is not a general thrift reader (it's only reading Parquet's thrift spec). So maybe leave a comment/TODO acknowledging that this check isn't done.